launcher/firstLaunch/firstlaunch_moc.cpp

```diff
@@ -296,29 +296,46 @@ QString FirstLaunchView::getHeroesInstallDir()
 void FirstLaunchView::extractGogData()
 {
 #ifdef ENABLE_INNOEXTRACT
-	QString filterExe = tr("GOG executable") + " (*.exe)";
-	QString fileExe = QFileDialog::getOpenFileName(this, tr("Select a GOG installer (exe) file..."), QDir::homePath(), filterExe);
+
+	auto fileSelection = [this](QString type, QString filter, QString startPath = {}) {
+		QString titleSel = tr("Select a ") + filter + tr(" file...");
+		QString titleErr = tr("You have to select a ") + filter + tr(" file!");
+#if defined(VCMI_MOBILE)
+		filter = "GOG file (*.*)";
+		QMessageBox::information(this, tr("File selection"), titleSel);
+#endif
+		QString file = QFileDialog::getOpenFileName(this, titleSel, startPath.isEmpty() ? QDir::homePath() : startPath, filter);
+		if(file.isEmpty())
+			return QString{};
+		else if(!file.endsWith("." + type, Qt::CaseInsensitive))
+		{
+			QMessageBox::critical(this, tr("Invalid file selected"), titleErr);
+			return QString{};
+		}
+
+		return file;
+	};
+
+	QString fileExe = fileSelection("exe", tr("GOG installer") + " (*.exe)");
```

Comment:
FYI in rare cases like this placeholder can be omitted, but still it might be an issue e.g. for RTL languages